library/game_state.py

```diff
@@ -155,9 +198,25 @@ def confirm_good_move(self, board: Board) -> bool:
 
         return self.good_move
 
+    def confirm_good_subgrid(self, board: Board) -> bool:
+        """Handle the entry of a bad subgrid choice"""
+        # TODO need to finish this. maybe move to game logic?
+        # guilty until proven innocent
+        self.good_move = False
+
+        working_subgrid = board.collect_subgrid(str(self.user_select_subgrid))
+        if board.check_grid_victory(working_subgrid) not in ("X", "O"):
+            self.good_move = True
+
+        return self.good_move

```

Comment:
this variable doesn't have to be attached to the class (`self.good_move`), can just be local to function.

Reply:
Removed the reference to self for good_move

Reply:
btw the function above it (forgot the name) has the same problem.